### DISCUSSION TITLE
PropertyChanged not working

### DISCUSSION THREAD

**Discussion by drei97** [AvaloniaUI/Avalonia#11295] [Q&A]
Hello,

in my project is a view (UserControl) using data binding. 

```
<TextBox Text="{Binding Entity.Datum, StringFormat={}{0:dd.MM.yyyy}}" />
```

where Entity is property in the viewmodel of the control

```
private AngebotEntity _entity;
public AngebotEntity Entity
{
      get => _entity;
      set => this.RaiseAndSetIfChanged(ref _entity, value);
}
```

and the DataContext is set properly

```
public partial class AngeboteView : UserControl
{  
    private AngeboteViewModel _viewmodel;  
    public AngeboteView()
    {
        InitializeComponent();
        DataContext = _viewmodel = new(this);
    }
}
```

When I modify a property of  Entity programmatically in the viewmodel

```
            Entity.Datum = res[0];
            this.RaisePropertyChanged("Entity");
```

the view is not updated. 

What's wrong?

**Comment by timunie** [comment]
Not entity has changed but `Datum`. so `Datum` needs to be an observable property that can raise PropertyChanged.

**Comment by drei97** [comment]
Since there is no Datum property I throw a PropertyChanged for the whole Entity which should give a notification to the view, which then is triggered to update the whole sub controls. That's the idea behind.

In the view all paths to the binded properies are like Entity.Datum. And if Entity has changed (due to the notification) all controls with a binding to Entity and its properties must be updated.

As far as I know this is how WPF is working. It seems that Avalonia is working differently?

Is it really necessary to have observable properties for all the Entity.Datum and other properties of Entity? Why then is a binding like Entity.Datum in a control possible if there is no way to notify the view about any change?

**Reply by timunie** [reply]
1. there are bindings that are just one time looked up, so no need for them to be observable. Anyway, probably it should also work as you said, but not 100 % sure. I'll take a deeper look if you provide a minimal sample App.

**Reply by drei97** [reply]
here it is
[notifychanged.zip](https://github.com/AvaloniaUI/Avalonia/files/11432000/notifychanged.zip)

**Reply by rabbitism** [reply]
If you don't want to implement INPC for DataModel, then you need to assign the entire Model property. Notify UI "Model" is changed cannot make a change to "Model.MyText"

**Reply by drei97** [reply]
INPC ?

**Reply by drei97** [reply]
I also have a view as DataGrid.

```
<DataGrid x:Name="datagrid" Items="{Binding Data}"
...
    <DataGrid.Columns>
        <DataGridTextColumn Header="Datum"
                            Binding="{Binding Datum, StringFormat={}{0:dd.MM.yyyy}}"/>                          

...
```

with

```
public ObservableCollection<AngebotEntity> Data
{
    get => _data;
    set => this.RaiseAndSetIfChanged(ref _data, value);
}
```

and

```
public void Call()
{
    Data = new ObservableCollection<AngebotEntity>(Controller.Angebote.OrderBy(xx => xx.Datum));
}
```

Here I do not have to create observable properties. A PropertyNotifyChanged of Data updates the whole view in once.

**Reply by drei97** [reply]
My latest post is not correct by 100%. If I add new data/rows and a RaisePropertyChanged updates the view.
But if I modify only a part like "Datum" the view is not updated.

How to update the view if only a single data field is modified by code behind?

**Reply by timunie** [reply]
Thank you for the sample. After exploring it, seems to be more like a bug to me tbh. 
- tested also CommunityToolkit.MVVM just to verify
- also tested 11.0.0-preview8 and issue is still there.
